src/hooks/Circles/usePrivatePosts.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  InfiniteData,
+  useInfiniteQuery,
+  useMutation,
+  useQueryClient,
+} from 'react-query';
+import { gql } from 'graphql-request';
+import { useGraphQLClient } from '../useGraphQLClient';
+import { useActiveAccount } from '../useActiveAccount';
+import { Post, postDetailsFragment } from './types';
+import { useUser } from '../useUser';
+import { IMessage } from 'react-native-gifted-chat';
+import uuid from 'react-native-uuid';
+
+/**
+ * privatePosts
+ */
+export type InfinitePrivatePostsData = InfiniteData<PrivatePostsData>;
+export type PrivatePostsData = {
+  privatePosts: {
+    pageInfo: {
+      endCursor: string;
+      hasNextPage: boolean;
+    };
+    edges: {
+      node: Post;
+    }[];
+  };
+};
+
+export type PostDetailsPostQueryResponse = {
+  post: Post;
+};
+
+export type PostRepliesQueryResponse = {
+  post: Pick<Post, 'replies'>;
+};
+
+export const postToMessage = (post: Partial<Post>): IMessage => ({
+  _id: post.id!,
+  text: post.message || '',
+  createdAt: post.createdAt ? new Date(post.createdAt) : new Date(),
+  user: {
+    _id: post.authorId!,
+    name: post.author?.profile.displayName! ?? 'Test?',
+    avatar: post.author?.profile.picture!,
```

Comment:
Is `picture` guaranteed to be present on `profile`?

Reply:
No, I will update this in the next PR.